hazelcast/src/test/java/com/hazelcast/map/EvictionTest.java

```diff
@@ -174,73 +176,135 @@ public void testMaxIdle_zeroIsInfinity() {
+        view = map.getEntryView(1);
+        expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);
+        delta = (1 + MILLISECONDS.toSeconds(sleepRefAfter - sleepRefBefore));
+        assertEquals(10, expected, delta);
         assertTrue(map.containsKey(1));
 
+        // Prolong 4th round
+        sleepRefBefore = System.currentTimeMillis();
         map.putAsync(1, "value4").get();
-        sleepAtLeastSeconds(2);
+        sleepRefAfter = System.currentTimeMillis();
+
+        view = map.getEntryView(1);
+        expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);
+        delta = (1 + MILLISECONDS.toSeconds(sleepRefAfter - sleepRefBefore));
+        assertEquals(10, expected, delta);
         assertTrue(map.containsKey(1));
 
+        // Prolong 5th round
+        sleepRefBefore = System.currentTimeMillis();
         map.setAsync(1, "value5").get();
-        sleepAtLeastSeconds(2);
+        sleepRefAfter = System.currentTimeMillis();
+
+        view = map.getEntryView(1);
+        expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);
+        delta = (1 + MILLISECONDS.toSeconds(sleepRefAfter - sleepRefBefore));
+        assertEquals(10, expected, delta);
         assertTrue(map.containsKey(1));
 
-        assertTrue(map.tryPut(1, "value6", 333, TimeUnit.MILLISECONDS));
-        sleepAtLeastSeconds(2);
+        // Prolong 6th round
+        sleepRefBefore = System.currentTimeMillis();
+        map.tryPut(1, "value6", 5, TimeUnit.SECONDS);
+        sleepRefAfter = System.currentTimeMillis();
+
+        view = map.getEntryView(1);
+        expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);
+        delta = (1 + MILLISECONDS.toSeconds(sleepRefAfter - sleepRefBefore));
+        assertEquals(10, expected, delta);
         assertTrue(map.containsKey(1));
 
+        // Prolong 7th round
+        sleepRefBefore = System.currentTimeMillis();
         map.replace(1, "value7");
-        sleepAtLeastSeconds(2);
+        sleepRefAfter = System.currentTimeMillis();
+
+        view = map.getEntryView(1);
+        expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);
+        delta = (1 + MILLISECONDS.toSeconds(sleepRefAfter - sleepRefBefore));
+        assertEquals(10, expected, delta);
         assertTrue(map.containsKey(1));
 
+        // Prolong 8th round
+        sleepRefBefore = System.currentTimeMillis();
         map.replace(1, "value7", "value8");
-        sleepAtLeastSeconds(2);
+        sleepRefAfter = System.currentTimeMillis();
+
+        view = map.getEntryView(1);
+        expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);
+        delta = (1 + MILLISECONDS.toSeconds(sleepRefAfter - sleepRefBefore));
+        assertEquals(10, expected, delta);
         assertTrue(map.containsKey(1));
+
+        // Confirm expiration
+        sleepAtLeastSeconds(10);
+        assertFalse(map.containsKey(1));
```

Comment:
I think this test may fail to detect broken methods. Since there is not an actual sleep, all these operations are likely to complete within one second. Since maxidle precision is seconds, `view.getExpirationTime()` will return the same second for all the calls in this test. Therefore, we would not know if one of the operations did not actually prolong max-idle. In fact, I put a 
```System.out.println(view.getExpirationTime());```
 before each
```expected = MILLISECONDS.toSeconds(view.getExpirationTime() - sleepRefBefore);```
 in this test. I got the following output on my local run:
```
1537822395000
1537822395000
1537822395000
1537822395000
1537822395000
1537822395000
1537822395000
1537822395000
1537822395000
```
This validates my concern about this way of testing.

Reply:
Thanks @mustafaiman - i missed the sleeps, you are absolutely right.